msg/templates/uorb_microcdr/microRTPS_client.cpp.em

```diff
@@ -105,13 +108,28 @@ void* send(void* /*unused*/)
 @[for idx, topic in enumerate(send_topics)]@
         @(send_base_types[idx])_s @(topic)_data;
         if (@(topic)_sub.update(&@(topic)_data)) {
+@[if topic == 'Timesync' or topic == 'timesync']@
+            if(@(topic)_data.sys_id == 0 && @(topic)_data.seq != last_remote_msg_seq && @(topic)_data.tc1 == 0) {
```

Comment:
I know you've done it here for the timesync, but in general what is the mechanism that stops a message that is sent into PX4 via the RTPS from being immediately sent back out again?

I feel like this isn't the only place where we are going to have this issue...

Reply:
@jkflying this is going to happen to all messages that are set to be sent and received simultaneously in the bridge. There won't be as much as I can think of. But the mechanism will always be associate an ID to identify the source of the message, and that way you know for sure you are not publishing the message back. If it makes sense to add a source ID by default to all uORB, probably not - what we probably need to make sure is if the user sets a message to be sent and received on the link, then it needs to have an identifier to the source.

Reply:
I will generalise this mechanism as soon as it is justifiable.

Reply:
Ok. I can immediately think of 2 other cases where we do this with MAVLink, OBSTACLE_DISTANCE (received is from a sensor, sent is the vehicle's internal representation of the fused sensors) and TRAJECTORY_WAYPOINT (received is setpoints to control the drone, sent is what would have been those setpoints if the avoidance wasn't enabled). 

Maybe there's a better way we can do this, structurally around how/when the messages get published, so that we don't get these loops in the future. Or just enforce that there is a sys_id in any message that gets both sent and received :man_shrugging:  I agree, doesn't have to be part of this PR though.